token.go

```diff
@@ -391,6 +392,7 @@ func (t *BearerToken) getHeader(name string) (string, error) {
 // There's a deeper explanation of this in the Vault source code.
 // https://github.com/hashicorp/vault/blob/b17e3256dde937a6248c9a2fa56206aac93d07de/builtin/credential/aws/path_login.go#L1569
 func buildHttpRequest(method, endpoint string, parsedUrl *url.URL, body string, headers http.Header) (*http.Request, error) {
+	endpoint = strings.TrimSuffix(endpoint, "/")
```

Comment:
Can you also add test for this change aswell.

Reply:
I included a test [below](https://github.com/hashicorp/consul-awsauth/pull/9/files#r1935936745) that demonstrates the issue